r/R/array.R

```diff
@@ -66,7 +66,14 @@
 #'
 #' @export
 array <- function(...){
-  `arrow::Array`$dispatch(Array__from_vector(vctrs::vec_c(...)))
+  args <- list(...)
+  args$type <- NULL
+
+  `arrow::Array`$dispatch(
+    Array__from_vector(
+      do.call("vec_c", args, envir = as.environment(asNamespace("vctrs")))
```

Comment:
Instead of `do.call`, we can splice in `vec_c`, e.g. 

``` r
library(vctrs)
data <- list(1:10, 1:10)
vec_c(!!!data)
#>  [1]  1  2  3  4  5  6  7  8  9 10  1  2  3  4  5  6  7  8  9 10
```

Reply:
Sorry, I thought it was handled somewhere since I thought it was a given that this was already supported and were just missing to build the `vctrs`. I'm actually not using this at all, so let me close this PR.